### ISSUE TITLE
add `availableProjects()` to SDK 

### ISSUE THREAD

**Issue by samuelstroschein** [inlang/monorepo#1814]
## Problem 

Every app needs to implement "available projects" logic e.g. what projects are available in a repository? 

## Proposal

Expose a function that apps can use to traverse repositories and find projects. 

- simplify building inlang apps because logic does not need to be re-implemented
- chances of less bugs due to one logic
- SDK could cache availableProjects in the future to increase start up times

```ts
import { loadProject, availableProjects } from "@inlang/sdk"

const projects = await availableProjects({ fs })
const userSelection: AvailableProject
const project = await loadProject({ projectPath: userSelection.projectPath, fs })

```

**Comment by samuelstroschein** [comment]
cc @flornkm @NiklasBuchfink 

manage.inlang.com needs this logic and so does the editor. What do you think?

**Comment by flornkm** [comment]
YES! Really good proposal. Didn't knew if the apps should look up all projects themselves or if the SDK does that.

`availableProjects` indicates that a project could also be "disabled" imo

Maybe `getProjects`? Or just `allProjects`?

**Comment by NiklasBuchfink** [comment]
That is why I asked in the Loom of the directory proposal how we deal with different project names. I thought this must be coming :D 

- `availableProjects` is more descriptive than `getProjects`
- `listProjects` is also known to developers

**Comment by flornkm** [comment]
@samuelstroschein yes can do that as it is blocking me anyway.

**Just to be clear with the scope:**

an async function should go through every directory in the repo (starting at the path the user gives like in your example) and find all project.inlang directories and return them?

**Comment by samuelstroschein** [comment]
@flornkm Is the scope clear with the pseudocode below?

```
1. Find all directories that end with .inlang (*.inlang) from the `from` path
2. Return a list of the found directories
```

```ts
export async function listProjects(args: { nodeishFs: NodeishFilesystem, from: string }): Array<{ projectPath: string }>
```

- returning an object enables us to add properties to found projects in the future
- this recursion will be very expensive. we need to add caching logic in the future. in the meantime, you got add a recursion limit of 3 or another heuristic

**Comment by flornkm** [comment]
@samuelstroschein thanks, sounds logical to me!
